src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1549,6 +1601,32 @@ private SolutionState WithTextDocumentState(TextDocumentState newDocument, bool
             return this.ForkProject(newProject);
         }
 
+        private SolutionState WithAnalyzerConfigDocumentState(AnalyzerConfigDocumentState newDocument, bool textChanged = false, bool recalculateDependentVersions = false)
+        {
+            if (newDocument == null)
```

Comment:
\<rant>
Times like this make me wish we could do `newDocument ??= throw new ArgumentNullException(nameof(newDocument))` or even just have language magic for `thrownull newDocument`
\</rant>

Reply:
You can write `newDocument = newDocument ?? throw ...` If you'd prefer that. Or complain to @333fred who reminded me why we don't really allow that.

Reply:
We specifically didn't want to allow that. `??=` is an _assignment_. If you're not assigning something, then don't use it. If you want a one-liner, just say `if (newDocument == null) throw new ArgumentNullException(nameof(newDocument))`